### ISSUE TITLE
Create ask-donation banner for MoFo sites

### ISSUE THREAD

**Issue by gideonthomas** [mozilla/donate.mozilla.org#1856]
The design ticket for this is #1855

We want to develop a popup ~~modal~~ banner...
> ~~...similar  to our current modal that asks site visitors for an email subscription, we need to build one to ask site visitors to donate to Mozilla~~

We want to add this to the following sites (for now):
- ~~https://advocacy.mozilla.org/privacynotincluded/~~ (filed separate issue)
- https://advocacy.mozilla.org/net-neutrality/
- https://changecopyright.org/
- https://internethealthreport.org/ (TBD)

Ideally we want a single ~~modal~~ banner (probably as a react component) that we can use for the first three sites but let's discuss in the design phase if that is possible.

The fourth site (Internet Health Report) is a little more challenging since it is a wordpress site. We might be a bit confined there in terms of what we can implement so that is a design consideration.

MoCo is also implementing this on mozilla.org, however, for engineering, we might not be able to reuse a lot of their code due to differing technologies. We can evaluate this once the design phase is done.

**Comment by Pomax** [comment]
Are all these sites React, or do we want a universal modal that can be used anywhere, with a React component for those web properties that would benefit from one?

**Comment by gideonthomas** [comment]
@Pomax, it depends on what you think is easier to implement.

Advocacy (https://github.com/mozilla/advocacy.mozilla.org) is in react.
Change copyright (https://github.com/mozilla/copyright) is also in react.

Internet Health Report is in Wordpress and I'm still waiting to hear the verdict on whether we are going to pursue adding it to that (I should find out by tomorrow).

**Comment by Pomax** [comment]
Also, where is a clear example of the "[...] current modal that asks site visitors for an email subscription"?

**Comment by gideonthomas** [comment]
Copy for the banner:

---

**We all love the web. Join Mozilla in defending it.** 
Let’s protect the world’s largest public resource for future generations. Just a few times a year, Mozilla asks for donations. If everyone reading this chipped in $3, we could wrap this fundraiser in under an hour. Together we can keep the web healthy, wonderful and welcoming to all.

\<donate button\>

---

**Comment by gideonthomas** [comment]
Sorry, that was for the `privacynotincluded` site. I'll PM you the design for the mozilla.org banner.

**Comment by gideonthomas** [comment]
Bedrock issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1415295

**Comment by Pomax** [comment]
It might be worth implementing this as an independent "library" which we can then import/require into any React codebase (to start with). Should the banner initiate a process that takes them to donate.mozilla.org, or should it drop them back on the page they were just on after completion?

**Comment by gideonthomas** [comment]
the "Donate" button in the banner should just link to the donate website. Don't have to handle after completion

**Comment by alanmoo** [comment]
Let's open a new issue this year if needed, based on where we are with our web properties.
